autoreporter_addons/ssl_checks/reporter.py

```diff
@@ -0,0 +1,203 @@
+                if redirect_url_parsed.hostname != task_result["payload"]["domain"]:
+                    return []
+
+        result = []
+        if task_result["result"].get("certificate_authority_invalid", False):
+            result.append(
+                Report(
+                    top_level_target=get_top_level_target(task_result),
+                    target=f'https://{task_result["payload"]["domain"]}:443/',
+                    report_type=SSLChecksReporter.CERTIFICATE_AUTHORITY_INVALID,
+                    additional_data={},
+                    timestamp=task_result["created_at"],
+                )
+            )
+        if task_result["result"].get("bad_redirect", False):
+            response_content_prefix = task_result["result"].get("response_content_prefix", "")
+            if not any(
+                [fragment in response_content_prefix for fragment in FILTERED_WEBSITE_FRAGMENTS_FOR_BAD_REDIRECT]
+            ):
+                # If there is some kind of HTML redirect, let's better not report that, as it might be
+                # a proper SSL redirect - here, we want to decrease the number of false positives at the
+                # cost of true positives.
+                try:
+                    soup = BeautifulSoup(response_content_prefix.lower(), "html.parser")
+                except Exception:  # parsing errors
+                    logger.exception("Unable to parse HTML from %s", task_result["payload"]["domain"])
+                    soup = None
+
+                if not soup or not soup.find_all("meta", attrs={"http-equiv": "refresh"}):
+                    result.append(
+                        Report(
+                            top_level_target=get_top_level_target(task_result),
+                            target=f'http://{task_result["payload"]["domain"]}:80/',
+                            report_type=SSLChecksReporter.NO_HTTPS_REDIRECT,
+                            additional_data={},
+                            timestamp=task_result["created_at"],
+                        )
+                    )
+
+        if task_result["result"].get("cn_different_from_hostname", False):
+            # If the domain starts with www. but the version without www. is in the names list,
+            # let's assume just the version without www. is advertised and used by the users.
+            if not (
+                task_result["payload"]["domain"].startswith("www.")
+                and task_result["payload"]["domain"][4:] in task_result["result"]["names"]
+            ):
+                result.append(
+                    Report(
+                        top_level_target=get_top_level_target(task_result),
+                        target=f'https://{task_result["payload"]["domain"]}:443/',
+                        report_type=SSLChecksReporter.BAD_CERTIFICATE_NAMES,
+                        additional_data={"names_string": ", ".join(sorted(set(task_result["result"]["names"])))},
```

Comment:
for the sake of code cleanness, I would put `", ".join(sorted(set(task_result["result"]["names"])))` in var